zerver/lib/narrow.py

```diff
@@ -957,7 +957,10 @@ def add_narrow_conditions(
 
     if search_operands:
         is_search = True
-        query = query.add_columns(topic_column_sa(), column("rendered_content", Text))
+        query = query.add_columns(
+            func.escape_html(topic_column_sa(), type_=Text).label("escaped_topic_name"),
+            column("rendered_content", Text),
+        )
```

Comment:
Can you add a block comment here explaining how the topic escaping logic works end-to-end? It's pretty important, and I'd like to make sure we address clearly questions like "Does `process_fts_updates` need to do something special to agree with this?"

Reply:
Added the comment in the code addressing these questions.